concent_api/core/transfer_operations.py

```diff
@@ -102,33 +99,64 @@ def store_pending_message(
     )
 
 
-def create_file_transfer_token(
+def create_file_transfer_token_for_concent(
+    report_computed_task: message.tasks.ReportComputedTask,
+    authorized_client_public_key: bytes,
+    operation: FileTransferToken.Operation,
+) -> FileTransferToken:
+    ten_minutes = 600
+    return _create_file_transfer_token(report_computed_task, authorized_client_public_key, operation, ten_minutes)
+
+
+def create_file_transfer_token_for_golem_client(
     report_computed_task:       message.tasks.ReportComputedTask,
     authorized_client_public_key: bytes,
     operation:                  FileTransferToken.Operation,
+) -> FileTransferToken:
+    return _create_file_transfer_token(report_computed_task, authorized_client_public_key, operation)
+
+
+def _create_file_transfer_token(
+    report_computed_task: message.tasks.ReportComputedTask,
+    authorized_client_public_key: bytes,
+    operation: FileTransferToken.Operation,
+    deadline: Optional[int] = None
+
 ) -> FileTransferToken:
     """
     Function to create FileTransferToken from ReportComputedTask message
     """
+    def calculate_token_expiration_deadline(
+        operation: FileTransferToken.Operation,
+        deadline: Optional[int],
+    ) -> int:
+        if deadline is not None:
```

Comment:
Extra assert that `operation in FileTransferToken.Operation` won't hurt, if that condition won't be met the code will fail.

Reply:
The situation when a condition cannot be met is exactly the case where an assert is appropriate. Otherwise you would use an `if`.

I would not call it useless duplication in the same sense the type annotation are not useless duplication.

Reply:
Basically, without assert if you can this function with `deadline = 1` and `operation = 'something'` the code will fail on line 157.

Reply:
how one should call this function directly when it is inner function of `_create_file_transfer_token` which already has this `assert`?

Reply:
It is now, but you never know how it is going to be used in the future. For example, I am going to use this function in verification too.

Reply:
@rwrzesien I just talked about this with @dybi
I missed that it's an inner function. In that case I'd say that skipping it is fine. No one is going to call it from the outside and the outer function already has an assert for that.

Reply:
:+1: